sdk/test/trace/tracer_test.cc

```diff
@@ -45,6 +45,24 @@ class MockSampler final : public Sampler
   nostd::string_view GetDescription() const noexcept override { return "MockSampler"; }
 };
 
+/**
+ * A Mock Custom Id Generator
+ */
+class MockIdGenerator : public IdGenerator
```

Comment:
nit: Google Mock would be great to use here, I noticed we don't have it in the repo but EXPECT_CALL and return values can all be done in the test body which would be nice